sys/mips/mips/support.S

```diff
@@ -999,9 +1000,14 @@ END(ffs)
  */
 LEAF(atomic_set_16)
 	.set	noreorder
-	srl	a0, a0, 2	# round down address to be 32-bit aligned
-	sll	a0, a0, 2
-	andi	a1, a1, 0xffff
+	/* NB: Only bit 1 is masked so the ll catches unaligned inputs */
+	andi	t0, a0, 2	# get unaligned offset
+	xor	a0, a0, t0	# align pointer
+#if _BYTE_ORDER == BIG_ENDIAN
+	xori	t0, t0, 2
+#endif
+	sll	t0, t0, 3	# convert byte offset to bit offset
+	sll	a1, a1, t0	# put bits in the right half
```

Comment:
Does this need the `andi` of `a1` to clear the upper bits before it is shifted?  In the case that the value is actually aligned, this will still set garbage bits in 16-31?

Reply:
The arguments are u16 so the calling convention should mean they're zero.